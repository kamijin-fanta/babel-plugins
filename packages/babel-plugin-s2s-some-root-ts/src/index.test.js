import plugin from './index';
import pluginTester from 'babel-plugin-tester';

pluginTester({
  plugin,
  pluginOptions: {
  },
  snapshot: true,
  tests: [
    {
      title: 'basic union',
      code: 'empty',
      pluginOptions: {
        input: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/examples/**/actionTypes.ts',
        output: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/types/action.ts',
        importName: 'Action',
        exportName: 'Action',
      },
    },
    {
      title: 'basic array',
      code: 'empty',
      pluginOptions: {
        input: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/examples/**/actionTypes.ts',
        output: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/types/action.ts',
        importName: 'Epic',
        exportName: 'Epics',
        exportType: 'array',
      },
    },
    {
      title: 'basic object',
      code: 'empty',
      pluginOptions: {
        input: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/examples/**/actionTypes.ts',
        output: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/types/action.ts',
        exportName: 'reducers',
        exportType: 'object',
      },
    },
    {
      title: 'equal path',
      code: 'empty',
      pluginOptions: {
        input: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/examples/Container1/actionTypes.ts',
        output: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/examples/Container1/action.ts',
        importName: 'reducer',
        exportName: 'reducers',
        exportType: 'object',
      },
    },
    {
      title: 'prefix / suffix',
      code: 'empty',
      pluginOptions: {
        input: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/examples/**/actionTypes.ts',
        output: 'packages/babel-plugin-s2s-action-root-ts/src/__fixtures__/basic/types/action.ts',
        importName: 'Action',
        exportName: 'Action',
        prefix: 'pre_',
        suffix: '_suf',
      },
    },
  ],
});

pluginTester({
  plugin,
  pluginOptions: {
  },
  tests: [
    {
      title: 'option error(input)',
      code: 'empty',
      output: 'empty',
      error: 'require input option',
      pluginOptions: {
        output: '.',
      },
    },
    {
      title: 'option error(output)',
      code: 'empty',
      output: 'empty',
      error: 'require output option',
      pluginOptions: {
        input: '.',
      },
    },
  ],
});
