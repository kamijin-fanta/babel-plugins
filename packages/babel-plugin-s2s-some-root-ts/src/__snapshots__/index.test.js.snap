// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic array 1`] = `
"
empty

      ↓ ↓ ↓ ↓ ↓ ↓

/***** Do not edit this file *****/
import { Epic as Container1 } from \\"../examples/container1/actionTypes\\";
import { Epic as Container2 } from \\"../examples/container2/actionTypes\\";

export const Epics = [Container1, Container2];
"
`;

exports[`basic object 1`] = `
"
empty

      ↓ ↓ ↓ ↓ ↓ ↓

/***** Do not edit this file *****/
import { Container1 } from \\"../examples/container1/actionTypes\\";
import { Container2 } from \\"../examples/container2/actionTypes\\";

export const reducers = {
  Container1,
  Container2
};
"
`;

exports[`basic union 1`] = `
"
empty

      ↓ ↓ ↓ ↓ ↓ ↓

/***** Do not edit this file *****/
import { Action as Container1 } from \\"../examples/container1/actionTypes\\";
import { Action as Container2 } from \\"../examples/container2/actionTypes\\";

export type Action = Container1 | Container2;
"
`;

exports[`equal path 1`] = `
"
empty

      ↓ ↓ ↓ ↓ ↓ ↓

/***** Do not edit this file *****/
import { reducer as . } from \\"./actionTypes\\";

export const reducers = {
  .
};
"
`;

exports[`prefix / suffix 1`] = `
"
empty

      ↓ ↓ ↓ ↓ ↓ ↓

/***** Do not edit this file *****/
import { Action as pre_Container1_suf } from \\"../examples/container1/actionTypes\\";
import { Action as pre_Container2_suf } from \\"../examples/container2/actionTypes\\";

export type Action = pre_Container1_suf | pre_Container2_suf;
"
`;
